docs/caching2.md

```diff
@@ -0,0 +1,100 @@
+
+For logged-in queries, there is an opt-in directive `@cacheable` that can be added to a query. When present - no access token will be propagated with the request, and the `no-cache` value will not be included - which will allow the CDN to cache this query. This is useful for queries that are not user-specific, and can be cached for all users.
+
+There is also an escape hatch - a query param in the URL: `?nocache=true` - that will force the CDN to bypass the cache and hit Metaphysics directly _despite_ a `@cacheable` directive. This can be useful for a privileged user to see an unpublished page (which we wouldn't want to cache).
+
+Setting/unsetting the `Cache-Control` header, and propagating any `max-age` is accomplished in a Relay middleware: [`System/Relay/middleware/cacheHeaderMiddleware.tsx`](System/Relay/middleware/cacheHeaderMiddleware.tsx).
+
+Un-setting the access token header for `@cacheable` queries is accomplished in a Relay middleware: [`System/Relay/createRelaySSREnvironment.ts`](System/Relay/createRelaySSREnvironment.ts).
+
+#### Logged out:
+
+By default, we typically cache all queries, unless otherwise configured with `cacheConfig: { force: true }` or `serverCacheTTL: 0`. How this works is:
+
+- User visits a page and Relay issues a GraphQL request.
+- Unless otherwise specified - this request will not contain an access token header _or_ a `no-cache` header value and so can be cached by the CDN (with an optional `max-age`).
+- As user navigates on the client in SPA mode, queries continue to be issued that can be cached by the CDN (unless otherwise specified).
+
+### Configuring Cache
+
+At the router level, we can configure things in a few different ways:
+
+1. Never cache, period:
+
+```tsx
+const myRouter = [
+  {
+    path: "/some-app",
+    // Tell relay network layer to never cache. This flows down to setting
+    // a `no-cache` header when making the CDN request, thus bypassing that cache as well.
+    cacheConfig: {
+      force: true,
+    },
+  },
+]
+```
+
+2. When logged out, never cache on the server, always cache on the client. This ensure users will always receive fresh data when they first visit a page, but once visited, page will be cached on the client for a given session. (Timely pages like `/auction/:id` apply this strategy.)
+
+```tsx
+const myRouter = [
+  {
+    path: "/some-app",
+    // Tell redis to never cache queries on the server. We still cache on
+    // client-side SPA transitions at the relay level (only during the session).
+    // Only applies while logged out!
+    // This flows down to setting a `no-cache` header when making the CDN request, thus bypassing that cache as well.
+    serverCacheTTL: 0,
+  },
+]
+```
+
+3. Configure the server-side TTL cache time for a given route:
+
+```tsx
+const myRouter = [
+  {
+    path: "/some-app",
+    // This flows down to a `max-age` value in the CDN request as well.
+    serverCacheTTL: 24 * 60 * 60, // 24 hours,
+  },
+]
+```
```

Comment:
We'll want to add a `4.` example showing how we  can annotate route-level queries (and query renderers) with the `@cacheable` directive